src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -321,6 +338,16 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
         }
     }
 
+    private boolean isProxyRequested() {
+        // HTTPS requests have uri without http scheme too
+        if (currentRequest.getMethod() == HttpMethod.CONNECT || sslEngine != null) {
```

Comment:
No, in this case you get a secure connection failed since the CONNECT is 
answered with a 400 bad request. A HTTPS request is path only too and 
has to be bypassed.

Am 19.07.2015 um 21:38 schrieb Jason Hoetger:
> In src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java
> <https://github.com/adamfisk/LittleProxy/pull/220#discussion_r34961244>:
>
>> @@ -321,6 +338,16 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
>>          }
>>      }
>>
>> +    private boolean isProxyRequested() {
>> +        // HTTPS requests have uri without http scheme too
>> +        if (currentRequest.getMethod() == HttpMethod.CONNECT || sslEngine != null) {
>
> Does this mean that HTTPS requests to the proxy will not be caught by
> this logic?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adamfisk/LittleProxy/pull/220/files#r34961244>.
>